### PR TITLE
[DOC] Link fixes

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: file.c [hunk 1]
```diff
@@ -7135,7 +7135,7 @@ const char ruby_null_device[] =
  *
  *  Note that file permissions are quite different from the _mode_
  *  of a file stream (\File object).
- *  See IO@Modes.
+ *  See IO.
```
Comment (peterzhu2118) [inline file.c:7138]: Is there a specific part we can link to rather than the generic IO page?

File: io.c [hunk 2]
```diff
@@ -11957,7 +11957,7 @@ io_s_foreach(VALUE v)
  *
  *  - {Open Options}[rdoc-ref:IO@Open+Options].
  *  - {Encoding options}[rdoc-ref:encodings.rdoc@Encoding+Options].
- *  - {Line Options}[rdoc-ref:IO@Line+Options].
+ *  - {Line IO}[rdoc-ref:IO@Line+IO].
```
Comment (peterzhu2118) [inline io.c:11960]: The `Line IO` section doesn't seem to talk about keyword arguments, the `sep` and `limit` are both positional arguments. Can we even pass these as keyword arguments?

Comment (BurdetteLamar) [reply io.c:11960]: Out of scope for this PR?

Comment (peterzhu2118) [reply io.c:11960]: The Line IO section (removed in ded895baa96cb9622fcd44d308fb48341565605b) looked like this:

```
 *  === Line Options
 *
 *  A number of \IO methods accept optional keyword arguments
 *  that determine how lines in a stream are to be treated:
 *
 *  - +:chomp+: If +true+, line separators are omitted; default is +false+.
```

This is now linking to a different section that is not about keyword arguments.

Comment (BurdetteLamar) [reply io.c:11960]: Should this simply be removed?

Comment (peterzhu2118) [reply io.c:11960]: Well we should keep the part about the `chomp` keyword (maybe just paste it inline), since that's actually useful information.

Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c [hunk 3]
```diff
@@ -12055,7 +12054,7 @@ io_s_readlines(VALUE v)
...
```
Comment (peterzhu2118) [inline io.c:12057]: Ditto here.

Comment (BurdetteLamar) [reply io.c]: Fixed.  This is _really_ careless.  The report showed both breaks.  Thanks for being a lert; we need more lerts.

**Comment by peterzhu2118** [comment]
@BurdetteLamar can you respond to the comment above about linking to the generic `IO` page?

File: file.c (hunk 1 above)
Comment (BurdetteLamar) [reply file.c:7138]: Sorry, I thought I'd responded here.  I looked, but didn't see as suitable target.

Comment (peterzhu2118) [reply file.c:7138]: Can you fix this one in a separate PR then? (Or alternatively delete this completely). I don't think asking the reader to read all of the docs of IO is particularly helpful.

Comment (BurdetteLamar) [reply file.c:7138]: Sure.  I see the problem, but not the solution.

**Comment by BurdetteLamar** [comment]
To be revisited.

**Pull request closed without merging**